### PR TITLE
[BFN] Update psu.py to process sigterm signal

### PR DESCRIPTION
**Pull request by dmytroxIntel**
#### Why I did it
Sometime, SIGTERM processing by psud takes more then default 10sec (please see stopwaitsecs in http://supervisord.org/configuration.html).

Due to this, the following two testcases may fail:

```
test_pmon_psud_stop_and_start_status
test_pmon_psud_term_and_start_status
```
#### How I did it
Update PSU plugin to process sigterm signal so that psud runs faster to end last cycle in time
#### How to verify it
Run SONiC CTs:
test_pmon_psud_stop_and_start_status
test_pmon_psud_term_and_start_status
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/psu.py [hunk 1]
```diff
@@ -267,23 +288,33 @@ def get_position_in_parent(self):
...
         """
         Retrieves the high threshold temperature of PSU
         Returns:
             A float number, the high threshold temperature of PSU in Celsius
             up to nearest thousandth of one degree Celsius, e.g. 30.125
         """
-        return self.get_thermal(0).get_high_threshold()
+        # Operation may take a few seconds to process, so if state is
+        # "termination signal", plugin doesn't perform this operation
+        if not Psu.sigterm:
```
Comment (akokhan) [inline platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/psu.py:315]: Can't we add similar check inside `cancel_on_sigterm` ?

**Review by akokhan — APPROVED** [review]

**Comment by dmytroxIntel** [comment]
@richardyu-ms, @StormLiangMS, @lguohan , please review and merge

**Comment by dmytroxIntel** [comment]
@richardyu-ms, @StormLiangMS, @lguohan , please review and merge

**Comment by dmytroxIntel** [comment]
@richardyu-ms, @StormLiangMS, @lguohan , please review and merge

**Comment by dmytroxIntel** [comment]
@prgeor , please review and merge

**Pull request merged**